src/api/analytics/index.tsx

```diff
@@ -0,0 +1,37 @@
+import ReactGA, { InitializeOptions } from 'react-ga'
+import { isMobile } from 'react-device-detect'
+export { pageview as viewPage, event as triggerEvent } from 'react-ga'
+
+function getCustomBrowserType(): string {
+  if (isMobile) {
+    return 'ethereum' in window ? 'mobileWeb3' : 'mobileRegular'
+  } else {
+    return 'web3' in window ? 'web3' : 'desktop'
+  }
+}
+
+interface InitAnalyticsParams {
+  trackingCode: string
+  options?: InitializeOptions
+}
+
+export function initAnalytics(params: InitAnalyticsParams): void {
+  const { trackingCode, options } = params
+  if (typeof trackingCode === 'string') {
+    ReactGA.initialize(trackingCode, options)
+
+    ReactGA.set({
+      customBrowserType: getCustomBrowserType(),
+    })
+  } else {
```

Comment:
? what is this for

Reply:
This is a new dimension. To segment by the 4 values you see in the function implementation. Got the inspiration from uni code. Not an expert in analytics